pkgs/development/compilers/avian/default.nix

```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub, zlib, jdk }:
+
+stdenv.mkDerivation {
+  name = "avian-1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "readytalk";
+    repo = "avian";
+    rev = "72c0547f4bdedc1ba8da2882317e3f758deab23f";
```

Comment:
The construct `rec { /* ... */ }` defines a so-called recursive set, where keys can refer to other keys in the set being defined; without `rec`, other keys won't be in scope. It is basically the same as doing
```nix
let self = {
  a = 10;
  b = self.a;
}; in self
```